pkg/allocator/cache.go

```diff
@@ -243,3 +247,9 @@ func (c *cache) numEntries() int {
 	defer c.mutex.RUnlock()
 	return len(c.nextCache)
 }
+
+func (c *cache) deleteEntries() {
+	c.foreach(func(id idpool.ID, key AllocatorKey) {
+		c.deleteEntryLocked(id, key)
+	})
+}
```

Comment:
It looks like `c.foreach` read-locks on each iteration. Since we are remove entries, should we be write-locking instead?